pkg/server/endpoints/middleware.go

```diff
@@ -113,6 +113,16 @@ func AgentAuthorizer(log logrus.FieldLogger, ds datastore.DataStore, clk clock.C
 			return st.Err()
 		}
 
+		internal := func(reason types.InternalDetails_Reason, format string, args ...interface{}) error {
```

Comment:
I'm not sure I see a good reason to return the detailed internal failure reason to the agent. The reason to return the permission denied reason to the agent is to help it make informed reasons about how to respond. I would expect an agent to treat all `codes.Internal` the same.

Unless we have good reason, I'd prefer not to return these details.

Reply:
Sounds good, I'll remove it.